src/azure-cli-core/azure/cli/core/telemetry/_telemetry.py

```diff
@@ -0,0 +1,102 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+import getpass
+import time
+import subprocess
+import platform
+import json
+import os
+import sys
+
+from applicationinsights import TelemetryClient
+from applicationinsights.exceptions import enable
+from azure.cli.core import __version__ as core_version
+from azure.cli.core.application import APPLICATION, ARGCOMPLETE_ENV_NAME
+
+client = {}
+data = []
+
+def init_telemetry():
+    try:
+        instrumentation_key = '02b91c82-6729-4241-befc-e6d02ca4fbba'
+
+        global client #pylint: disable=global-statement
+        client = TelemetryClient(instrumentation_key)
+
+        client.context.application.id = 'Azure CLI'
+        client.context.application.ver = core_version
+        client.context.user.id = hash(getpass.getuser())
+
+        enable(instrumentation_key)
+    except Exception: #pylint: disable=broad-except
+        # Never fail the command because of telemetry
+        pass
+
+def user_agrees_to_telemetry():
+    # TODO: agreement, needs to take Y/N from the command line
+    # and needs a "skip" param to not show (for scripts)
+    return True
+
+# Note: do not log events with quotes in the name, properties or measurements;
+# those events may fail to upload.
+def log_event(event_name, value=None, measurements=None):
+    event_name = _remove_quotes(event_name)
+    value = _remove_quotes(value)
+    measurements = _remove_quotes(measurements)
+
+    props = {
+        'time': str(time.time()),
+        'correlation': APPLICATION.session['headers']['x-ms-client-request-id'],
```

Comment:
My suggestion would be to use the same name (i.e. x-ms-client-request-id) as other telemetry sources if possible.